src/commands/registries/genericV2/addTrackedGenericV2Registry.ts

```diff
@@ -8,7 +8,17 @@ import { ext } from "../../../extensionVariables";
 import { UnifiedRegistryItem } from "../../../tree/registries/UnifiedRegistryTreeDataProvider";
 
 export async function addTrackedGenericV2Registry(context: IActionContext, node?: UnifiedRegistryItem<unknown>): Promise<void> {
-    await ext.genericRegistryV2DataProvider.addTrackedRegistry();
+    const root = ext.genericRegistryV2DataProvider.getRoot();
+    const registries = await ext.genericRegistryV2DataProvider.getRegistries(root);
+
+    if (registries?.length > 0) {
+        // if there are already registries, add a new registry to the existing root node
+        await ext.genericRegistryV2DataProvider.addTrackedRegistry();
+    } else {
+        // if there are no registries, connect as usual
+        await ext.registriesTree.connectRegistryProvider(ext.genericRegistryV2DataProvider);
```

Comment:
Is `connectRegistryProvider` idempotent? If so we could just unconditionally call it. If not then maybe it should be...

Reply:
yeah i do like that more. we can do this by adding a condition here that checks if `provider` is `GenericV2RegistryProvider`, which is the edge case here. for all other cases, we shouldn't call `onConnect` again, but for GenericV2, we can. 

https://github.com/microsoft/vscode-docker/blob/f53de2c6b1cb47c563a81ece4b398a15d07b6324/src/tree/registries/UnifiedRegistryTreeDataProvider.ts#L164

another approach is to just remove the if statement entirely, which would rely on the quickpicks above to not show registry providers that are already connected, (which i think is reliable enough)